### PR TITLE
Bug/translated topography

### PR DESCRIPTION
**Pull request by sannant**

### CODE REVIEW

**Review by pastewka — APPROVED** [review]

**Comment by sannant** [comment]
```
def test_translate_setter():
    topography = Topography(np.array([[0, 1, 0], [0, 0, 0]]),
                            physical_sizes=(4., 3.))

    translated_t = topography.translate()

    translated_t.offset = (1, 0)
    assert (translated_t.heights()
            ==
            np.array([[0, 0, 0],
                      [0, 1, 0]])).all()
    topography = Topography(np.array([[0, 1, 0], [0, 0, 0]]),
                            physical_sizes=(4., 3.), periodic=True)
    assert topography.translate(offset=(1, 0)).is_periodic

```
this previously existing test  implies that it is ok to translate nonperiodic topographies (the first part, since by default, topographies are nonperiodic).

I think that this is bad behavior because the translation is implemented via a numpy.roll.

Do we agree to raise an error when a nonperiodic topography is translated ?

**Comment by sannant** [comment]
@pastewka I need your opinion on the comment above

**Comment by pastewka** [comment]
Agree to raise the error. Can you indicate in the changelog that this is an API change (prefix with API, not changelog) which triggers a new major version release.

**Comment by sannant** [comment]
Ok

File: SurfaceTopography/ChangeLog.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 Change log for SurfaceTopography
 ================================
 
+v1.12.3 (13Jan24)
```
Comment (pastewka) [inline SurfaceTopography/ChangeLog.md:4]: Can you bump version to v1.13.0 because this is a breaking change?

**Review by pastewka — CHANGES_REQUESTED** [review]

**Review by pastewka — APPROVED** [review]

**Pull request merged**